Simplenote/Classes/UITableView+Simplenote.swift

```diff
@@ -46,3 +46,120 @@ extension UITableView {
         return dequeueReusableHeaderFooterView(withIdentifier: T.reuseIdentifier) as? T
     }
 }
+
+// MARK: - Navigation
+//
+extension UITableView {
+
+    /// Returns first index path from the first non-empty section
+    ///
+    var firstIndexPath: IndexPath? {
+        let indexPath = IndexPath(row: 0, section: 0)
+        if numberOfRows(inSection: 0) == 0 {
+            return nextIndexPath(after: indexPath)
+        }
+
+        return indexPath
+    }
+
+    /// Returns index path after currently selected index path
+    ///
+    var nextIndexPath: IndexPath? {
+        guard let indexPath = indexPathForSelectedRow else {
+            return nil
+        }
+
+        return nextIndexPath(after: indexPath)
+    }
+
+    /// Returns index path before currently selected index path
+    ///
+    var prevIndexPath: IndexPath? {
+        guard let indexPath = indexPathForSelectedRow else {
+            return nil
+        }
+
+        var row = indexPath.row
```

Comment:
Nipticky: Maybe extract this, for symmetry?

```
    func indexPath(before indexPath: IndexPath) -> IndexPath?
```

Reply:
I like it, done!